std/algorithm/iteration.d

```diff
@@ -4852,37 +4905,101 @@ private struct UniqResult(alias pred, Range)
     void popFront()
     {
         assert(!empty, "Attempting to popFront an empty uniq.");
-        auto last = _input.front;
-        do
+        if (_input.empty)
         {
-            _input.popFront();
+            _isInFront = false;
+            static if (isBidirectionalRange!Range)
+            {
+                if (_isOverlapping)
+                {
+                    _isInBack = false;
+                }
+            }
+        }
+        else
+        {
+            _isInFront = true;
+            _front = _input.front;
```

Comment:
I wonder how copying to the new _front member will affect performance. Also, would it affect ranges with non-copyable elements?

PickPolicy does not look attractive anymore if it complicates the design.
Maybe we can .save the original range when moving backward, to fix the original issue; .back() can iterate towards front, saving the original range state and produce the result from the saved range's .back when the condition is not met. Would that work? I think it would remove the need for _front and other new state variables.